### PR TITLE
Add Yarn

### PR DESCRIPTION
**Pull request by gemmaleigh**
#### What does it do?
Add a yarn lockfile.

_In order to get consistent installs across machines, Yarn needs more
information than the dependencies you configure in your package.json.
Yarn needs to store exactly which versions of each dependency were
installed. To do this Yarn uses a yarn.lock file in the root of your project._

#### Migrating from NPM
https://yarnpkg.com/en/docs/migrating-from-npm

Other developers on the project can *keep using npm*, so you don’t need to get everyone on your project to convert at the same time. The developers using yarn will all get exactly the same configuration as each other, and the developers using npm may get slightly different configurations, which is the intended behavior of npm.

#### Travis CI and Yarn

[Travis CI supports Yarn](https://blog.travis-ci.com/2016-11-21-travis-ci-now-supports-yarn). If a yarn.lock file is found in the repository root, Travis will run `yarn` instead of `npm install`.

#### What type of change is it?
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
   "scripts": {
     "start": "gulp fractal",
     "lint": "standard",
-    "test": "npm run lint && gulp test"
+    "test": "yarn run lint && gulp test"
```
Comment (robinwhittleton) [inline package.json]: This means that regardless of whether they want to use npm or yarn, they at least have to have yarn installed to run the test suite right? Maybe we should get rid of the separate `lint` task for npm and change this to `standard && gulp test`, or use [gulp-standard](https://github.com/emgeee/gulp-standard)?

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,18 @@
 language: node_js
 node_js:
 - 6.6.0
+sudo: false
 before_install:
-- npm install -g mocha
-- npm install -g gulp
+  - curl -o- -L https://yarnpkg.com/install.sh | bash
```
Comment (robinwhittleton) [inline .travis.yml]: I’m a bit confused as to why we need this. Was under the impression that [Travis would use Yarn automatically](https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Travis-CI-supports-yarn) if `package.json` and `yarn.lock` were found.

**Comment by dsingleton** [comment]
I'm 👍  to `yarn`, i much prefer it's `bundle` style locking approach.

We had discussed using `yarn` back when it came out, but decided against it, thinking it might make contribution more difficult for folks new to the project. 

Now `yarn` has been out a little bit, people are more familiar with it, and it's more or less backwards compatible with `npm` use (the exception being someone adding a new dep with npm, and it'll get updated in the lockfile the next time someone installs using yarn, which'll need committing).

Overall, lets :shipit: and see how it works for us. This is the kind of project where we should be able to experiment with the tech a little :)

File: .travis.yml (hunk 2 above)
Comment (gemmaleigh) [inline .travis.yml]: I was under that impression too, Travis was still using npm for the build prior to this commit _possibly as yarn.lock isn't merged into master?_

File: package.json (hunk 1 above)
Comment (gemmaleigh) [inline package.json]: I'll update to use `standard && gulp test`, removing the yarn requirement.

File: .travis.yml (hunk 2 above)
Comment (gemmaleigh) [inline .travis.yml]: https://yarnpkg.com/en/docs/install-ci#travis-tab, prior to adding this before_install line - the build failed on `yarn global add mocha`.

**Pull request merged**